### PR TITLE
Fix get_kit_api: budget was sized from API content, not the MCP tool-result limit

### PR DESCRIPTION
**Pull request by gtanczyk**
## Summary

`get_kit_api` (merged in #710, a few hours ago) is unusable in production right now — every call fails outright. This is a hotfix.

## Root cause

`DEFAULT_MCP_DIGEST_MAX_BYTES` was set to `100_000` by reasoning purely from the API content's own size ("covers shell + full current API + the note reserve"). That's a real constraint, but the wrong one — the constraint that actually matters is the calling **MCP client's ceiling on a single tool result**, which the server has no way to query and no say over.

Deployed against the real kit (`engineRef fb0cd30df1…`, the current `main` on the games repo), the default digest ran **~98,730 characters / ~27,600 tokens** (measured with `tiktoken` `cl100k_base` as a proxy), and a live client refused the tool result outright as exceeding its token ceiling. `get_kit_api` was completely unusable — a worse failure than the truncation bug it replaced, since that one degraded gracefully with an "Omitted for length" note and this one returns nothing at all.

## Fix

- Reproduced the exact production digest locally (same `engineRef`) and re-derived the budget from the real constraint instead of the content: **50,000 raw digest bytes measures ~15,800 tokens** against the real kit — comfortable margin under a ~25,000-token single-result ceiling, while still keeping ~30 KiB of live API content. The exemplar/audio/module-catalog shell is untouched at any budget; only the API section shrinks.
- Corrected `get_kit_api`'s and `get_kit`'s tool descriptions, which claimed "the whole reference in one call." Omission is now the expected common case for a real kit, not a rare edge case — the description says so and points agents at the browse tools (`list_kit_files` / `search_kit_files` / `read_kit_file`) for anything cut.
- Added a regression test (`kit-digest.test.ts`) that rebuilds an API section at the real kit's rough byte scale (~120 declarations, comparable to the real ~114) and asserts the default-budget MCP response stays under a byte proxy for the token ceiling. Verified it fails against the old `100_000` value (81,219 bytes vs. a 65,000-byte ceiling) before restoring the fix.
- Updated the `byoca-mcp` skill doc with the incident, the corrected reasoning, and an explicit note for whoever touches this constant next: size it from a real digest run through `compactKitDigestForApi` and measured in real tokens, not from `shared/game-kit.d.ts`'s byte count.

## Why this shipped undetected

No test in the original PR (#710) exercised realistic content size — the tests used small synthetic fixtures, so a "covers the full API" budget looked correct on paper and passed review on both repos. This PR's regression test specifically closes that gap by testing at the real kit's byte scale.

## Test plan

- [x] `apps/api` full suite: 185 files / 2,888 tests pass
- [x] `npm run lint` (eslint + comment-prose): clean
- [x] Typecheck: clean
- [x] New regression test verified to fail against the old (broken) `100_000` value, and pass against the fix
- [x] Reproduced the exact production failure locally against the real merged `engineRef`

https://claude.ai/code/session_0155jrUy76vVwwcdPLm2Vh3u

---
_Generated by [Claude Code](https://claude.ai/code/session_0155jrUy76vVwwcdPLm2Vh3u)_

### CODE REVIEW

**Comment by gtanczyk** [comment]
@codex

File: apps/api/src/mcp-server.ts [hunk 1]
```diff
@@ -2546,9 +2546,9 @@ export async function registerMcpServerRoutes(app: FastifyInstance, options: Mcp
         'This platform and its Creator Kit are not on the public web — an unanswered question ' +
         'about what it can build (multiplayer, persistent worlds, party games, …) is answered by ' +
         'get_kit_api or browse, never by web search. ' +
-        'For the API itself: get_kit_api (with this engineRef) for the whole prompt-ready reference ' +
-        "in one call, or the browse tools named in this reply's browse block (list/search/read) to " +
-        'read specific kit files instead — do not pull the whole kit into context. ' +
+        'For the API itself: get_kit_api (with this engineRef) for a prompt-ready orientation in ' +
+        "one call — it names anything it had to cut, use the browse tools named in this reply's " +
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/mcp-server.ts]: Confirmed and fixed in 5cf5d80. You're right — `elideDeclaration`'s member-wise trim only counts what it drops (`… N more members omitted`), it doesn't name it, unlike `selectApiBlocks`'s whole-declaration omission note. "Names anything it had to cut" overclaimed the guarantee for that case.

Reworded both `get_kit`'s and `get_kit_api`'s descriptions to distinguish the two: named omissions for whole declarations dropped, count-only for members trimmed out of a kept-but-abridged declaration — both still pointing at the browse tools or `shared/game-kit.d.ts` either way.

---
_Generated by [Claude Code](https://claude.ai/code)_
